jerry-ext/args/arg-transform-functions.c

```diff
@@ -0,0 +1,280 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript-ext/args.h"
+#include "jerryscript.h"
+
+/**
+ * Pop the current JS argument from the iterator.
+ * It will change the index and js_arg_p value in the iterator.
+ *
+ * @return the current JS argument.
+ */
+jerry_value_t
+jerryx_arg_js_iterator_pop (jerryx_arg_js_iterator_t *js_arg_iter_p) /**< the JS arg iterator */
```

Comment:
note: I've moved the push and pop functions into this source file only because I've previously deleted the arg-iterator-helper.c file during my experiments and was too lazy to re-add it. right now, no bias towards any approach (keep it here or split it out).